### ISSUE TITLE
Setup up Module for Android Build

### ISSUE THREAD

**Issue by BigPun86** [idehub/iab-android-test#1]
Hey there, how do you setup up this module to work on Android? I mean, what is/are the setups for settings.gradle, build.gradle and MainActivity?

Best Regards

**Comment by cbrevik** [comment]
Hey!

When I release this as an npm module I'll clean up a lot, but quick and dirty how to use;

Clone this repo to a folder outside your RN project. 
Add `"react-native-inapp-billing": "file:../iabtest/"` to your project.json dependencies, then do an `npm install`. It'll copy what is necessary to your `node_modules`.

Next in your MainActivity.java you add `import com.idehub.IabTest.InAppBillingBridgePackage;`, and add the module to the ReactInstanceManager with `.addPackage(new InAppBillingBridgePackage("YOUR_GOOGLE_PLAY_KEY", null, this))` Obviously substitute `YOUR_GOOGLE_PLAY_KEY` with your actual license key from Google Play.

In your javascript you'll need to import the module `const InAppBilling = require('react-native-inapp-billing');` and use it with for example 

```
InAppBilling.getPurchaseListingDetails('YOUR_SKU').then((details) => {
    console.log(details);
});
```

I'll be cleaning up this module in the next few days and release it as a real package though, so expect the JS/Java API integration to change.

**Comment by BigPun86** [comment]
I didn´t clone the project outside but instead did this: 

<img width="461" alt="screen shot 2016-01-12 at 18 10 48" src="https://cloud.githubusercontent.com/assets/7880624/12270750/e9ff4380-b957-11e5-87e7-deda6bd79848.png">

Does it make any difference? The node_module project structure seems to be correct:

![screen shot 2016-01-12 at 18 12 23](https://cloud.githubusercontent.com/assets/7880624/12270795/24dd61a8-b958-11e5-9625-a9b99f30ac03.png)

Now i imported InAppBillingBridgePackage into MainActivity as well as .addPackage etc.
Now i still can´t build...I think i have to set up settings.gradle: 

```
include ':InAppBilling', ':app'
project(':InAppBilling').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-inapp-billing')
```

 and build.gradle:

```
dependencies {
....
    compile project(':InAppBilling')
}
```

After doing this i get following error when trying to build:

<img width="776" alt="screen shot 2016-01-12 at 18 17 00" src="https://cloud.githubusercontent.com/assets/7880624/12270908/c9408c16-b958-11e5-9553-c0a35d3199a3.png">

**Comment by BigPun86** [comment]
Okay fixed it by adding **/android** to projectDir, which should be obvious to me actually :) 

```
include ':react-native-inapp-billing', ':app'
project(':react-native-inapp-billing').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-inapp-billing/android')
```

Now the Android Project is being built, but crashes as soon as i start the app :(

But i think this has nothing to do with your module:

![screenshot_2016-01-12-19-12-34](https://cloud.githubusercontent.com/assets/7880624/12272391/9090b546-b960-11e5-81ed-590c6ae6ccc3.png)

**Comment by cbrevik** [comment]
Totally forgot about the gradle-config, of course it'd be something like that.

Can you do a `adb logcat` in terminal and see if you can find any stack trace / exceptions?

**Comment by cbrevik** [comment]
This was closed via DM on Reactiflux (Discord), fixed by dropping RN version down to 0.15 in build.gradle.
